### PR TITLE
Add type definition for @rgrove/parse-xml.

### PR DESCRIPTION
**Pull request by petejohanson**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them. (See https://github.com/rgrove/parse-xml/pull/4 discussion)
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project. (I've attempted to, but it doesn't seem to want to let me do that for scoped packages like this)
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/rgrove__parse-xml/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "adjacent-overload-signatures": false,
```
Comment (DanielRosenwasser) [inline types/rgrove__parse-xml/tslint.json]: You shouldn't need these tslint rules to be disabled.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
LGTM but please enable linting

**Comment by petejohanson** [comment]
@DanielRosenwasser Removed all the tslint rules getting disabled, and then fixed up the various errors, which including renaming some of the types, but the core structure/hierarchy is unchanged.

Thanks.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you!! 😄

**Comment by petejohanson** [comment]
@DanielRosenwasser Thank you! How long does it usually take for these to get published to the NPM registry?

**Comment by DanielRosenwasser** [comment]
I believe packages are published on an hourly basis: https://github.com/Microsoft/types-publisher/issues/384

**Pull request merged**